src/test/scala/it/pagopa/interop/purposeprocess/PurposeApiServiceSpec.scala

```diff
@@ -1512,6 +1513,192 @@ class PurposeApiServiceSpec extends AnyWordSpecLike with SpecHelper with Scalate
+          riskAnalysis = Some(
+            PurposeManagementDependency.PurposeVersionDocument(documentId, "application/pdf", path, SpecData.timestamp)
+          ),
+          firstActivationAt = Some(SpecData.timestamp)
+        )
+
+      mockVersionFirstActivation(purposeId, purposeVersionId3, eService.producerId, purpose.consumerId, updatedVersion)
+      mockFileManagerStore("whateverPath")
+
+      (() => mockDateTimeSupplier.get()).expects().returning(SpecData.timestamp).once()
+
+      mockClientStateUpdate(purposeId, purposeVersionId3, AuthorizationManagementDependency.ClientComponentState.ACTIVE)
+
+      val expected: PurposeVersion = purposeVersion.toApi
+
+      Get() ~> service.createPurposeVersion(purposeId.toString, seed) ~> check {
+        status shouldEqual StatusCodes.OK
+        responseAs[PurposeVersion] shouldEqual expected
+      }
+    }
+
+    "succeed in case of version already published but goes to Waiting for Approval state" in {
+
+      val consumerId        = UUID.randomUUID()
+      val documentId        = UUID.randomUUID()
+      val purposeId         = UUID.randomUUID()
+      val purposeVersionId1 = UUID.randomUUID()
+      val purposeVersionId2 = UUID.randomUUID()
+      val purposeVersionId3 = UUID.randomUUID()
+      val eserviceId        = UUID.randomUUID()
+      val descriptorId      = UUID.randomUUID()
+
+      implicit val context: Seq[(String, String)] =
+        Seq("bearer" -> bearerToken, USER_ROLES -> "admin", ORGANIZATION_ID_CLAIM -> consumerId.toString)
+
+      val version2_1 = SpecData.purposeVersion.copy(id = UUID.randomUUID(), state = Active, dailyCalls = 1000)
+      val purpose2 = SpecData.purpose.copy(eserviceId = eserviceId, consumerId = consumerId, versions = Seq(version2_1))
+
+      val path: String                               = "/here/there/foo/bar.pdf"
+      val document: PersistentPurposeVersionDocument =
+        PersistentPurposeVersionDocument(documentId, "application/pdf", path, SpecData.timestamp)
+
+      val version1_1 = SpecData.purposeVersion.copy(id = purposeVersionId1, state = Active)
+      val version1_2 = SpecData.purposeVersion.copy(
+        id = purposeVersionId2,
+        state = WaitingForApproval,
+        riskAnalysis = Some(document),
+        firstActivationAt = Some(SpecData.timestamp)
+      )
+
+      val purpose =
+        SpecData.purpose.copy(
+          id = purposeId,
+          versions = Seq(version1_1, version1_2),
+          consumerId = consumerId,
+          eserviceId = eserviceId
+        )
+
+      val purposes                 = Seq(purpose, purpose2)
+      val seed: PurposeVersionSeed = PurposeVersionSeed(dailyCalls = 1000)
+
+      val purposeVersion = PersistentPurposeVersion(
+        id = purposeVersionId3,
+        state = WaitingForApproval,
+        createdAt = SpecData.timestamp,
+        updatedAt = None,
+        expectedApprovalDate = None,
+        dailyCalls = seed.dailyCalls,
+        riskAnalysis = Some(document),
```

Comment:
as per the other comment
```suggestion
        riskAnalysis = None,
```